templates/app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class CartsController < StoreController
+  helper 'spree/products', 'orders'
+
+  respond_to :html
+
+  before_action :store_guest_token
+  before_action :assign_order, only: :update
+  # note: do not lock the #edit action because that's where we redirect when we fail to acquire a lock
+  around_action :lock_order, only: :update
+
+  # Shows the current incomplete order from the session
+  def edit
+    @order = current_order(build_order_if_necessary: true)
+    authorize! :edit, @order, cookies.signed[:guest_token]
+    if params[:id] && @order.number != params[:id]
+      flash[:error] = t('spree.cannot_edit_orders')
+      redirect_to edit_cart_path
+    end
+  end
+
+  def update
+    authorize! :update, @order, cookies.signed[:guest_token]
+    if @order.contents.update_cart(order_params)
+      @order.next if params.key?(:checkout) && @order.cart?
+
+      respond_with(@order) do |format|
+        format.html do
+          if params.key?(:checkout)
+            redirect_to checkout_state_path(@order.checkout_steps.first)
+          else
+            redirect_to edit_cart_path
+          end
+        end
+      end
+    else
+      respond_with(@order)
+    end
+  end
+
+  def empty
+    if @order = current_order
+      authorize! :update, @order, cookies.signed[:guest_token]
+      @order.empty!
+    end
+
+    redirect_to edit_cart_path
+  end
+
+  private
+
+  def accurate_title
+    t('spree.shopping_cart')
```

Comment:
I removed the conditional when I copied `accurate_title` method from OrdersController to CartsController.